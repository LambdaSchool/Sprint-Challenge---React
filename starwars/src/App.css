* {
	box-sizing: border-box;
}

.App {
	max-width: 1600px;
	width: 100%;
	margin: 0 auto;

}

.Header {
	color: #443e3e;
	text-shadow: 1px 1px 5px #fff;
	font-size: 4rem;
	text-align: center;
}

/* write your parent styles in here for your App.js */

.Starwars {
	display: flex;
	flex-wrap: wrap;
	justify-content: center;
	margin: 0 auto;
	align-items: center;
	width: 100%;
}

.CharacterList {
	max-height: 500px;
	border: 1px solid black;
	width: 30%;
	background-color: #3C6478;
	margin: 20px 3% 20px 0;
	border-radius: 10px;
	color: white;
}

.CharacterList:hover {
	background-color: #E5D469;
	color: black;
}

.characterCards {
	font-size: 1.4rem;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	text-align: center;
	text-shadow: 1px 1px 5px #fff;
}

.cardHeader {
	color: white;
	font-size: 1.2rem;
	text-shadow: 1px 1px 5px #fff;
}
