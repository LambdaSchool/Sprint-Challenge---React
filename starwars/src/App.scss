.App {
  text-align: center;
}
.Header {
  color: #443e3e;
  text-shadow: 1px 1px 5px #fff;
  font-size: 5.2rem;
}
/* write your parent styles in here for your App.js */

@import './variables';

.next-button {

  position: fixed;
  top: 50%;
  right: 10px;
  border: 1px solid black;
  background-color: white;
  padding: 20px 40px;
  border-radius: 40%;
  cursor: pointer;

  &:hover {

    background-color: #ccc;

  }

  @media (max-width: $mobile) {

    padding: 20px 10px;

  }

}

.previous-button {

  position: fixed;
  top: 50%;
  left: 10px;
  border: 1px solid black;
  background-color: white;
  padding: 20px 40px;
  border-radius: 40%;
  cursor: pointer;

  &:hover {

    background-color: #ccc;

  }

  @media (max-width: $mobile) {

    padding: 20px 10px;

  }

}

.load {

  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  padding: 30px;
  background-color: rgba(147, 104, 16, 0.5);
  width: 315px;
  border: 1px solid black;
  border-radius: 10px;
  color: white;

}

.invisible {

  display: none;

}
