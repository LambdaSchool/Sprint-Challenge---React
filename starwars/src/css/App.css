.App {
  text-align: center;
}
.Header {
  color: #443e3e;
  text-shadow: 1px 1px 5px #fff;
}
/* write your parent styles in here for your App.js */
.character-container {
  width: 100%;
  
}

.character-container:nth-child(odd) {
  margin-left: 300px;
}