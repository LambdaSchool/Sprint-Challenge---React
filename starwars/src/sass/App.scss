.App {
  text-align: center;
}
.Header {
  color: #443e3e;
  text-shadow: 1px 1px 5px #fff;
}

.board {
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  border: 21px solid red; // TEST: REPLACE ME WITH BOARD STYLES IN APP.CSS


  &-row:after {
    clear: both;
    content: '';
    display: table;
  }

  
  &-status {
    display: flex;
    flex-flow: column wrap;
    align-items: center;
    box-shadow: 1px 2px 1px rgba(200,200,200,0.8);
    margin: 2rem;
    padding: 1rem;
  }
}


.scoreboard {
  padding: .5rem;

  #p1 {
      color: blue;
      margin-right: 2rem
  }

  #p2 {
      color: red;
      margin-left: 2rem;
  }
}
