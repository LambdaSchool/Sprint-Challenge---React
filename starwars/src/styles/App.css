.cards .cards__row1 {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin-top: 50px;
}
.cards .cards__container {
  border: 2px solid #8a8a89;
  width: 40%;
  height: auto;
}
.App {
  text-align: center;
}
.App #logo {
  margin-top: 50px;
}
.Header {
  color: #443e3e;
  text-shadow: 1px 1px 5px #fff;
}
