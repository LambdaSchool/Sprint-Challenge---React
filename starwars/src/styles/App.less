@import "./cards.less";

.container {
  box-sizing: border-box;
  font-size: 62.5%;
  display: flex;
  
  .container__logo {
   margin-top: 50px;
   
  }

  .App__cards {
    .cards__row1 {
      margin-top: 50px;
    }
  }
}

.Header {
  color: #443e3e;
  text-shadow: 1px 1px 5px #fff;
}
