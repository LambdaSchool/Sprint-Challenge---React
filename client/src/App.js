import React, { Component } from "react";
import "./style/App.css";
import Card from "./components/card";
import { Port, Api } from "./variables";
import axios from "axios";

class App extends Component {
  constructor() {
    super();
    this.state = {
      starwarsChars: []
    };
  }

  componentDidMount() {
    axios
      .get(Api.REACT_APP_CHAR)
      .then((response) => {
        this.setState(() => ({ starwarsChars: response.data }));
      })
      .catch((error) => {
        console.error("Server Error", error);
      });
  }

  render() {
    console.log("this.state inside app.js", this.state.starwarsChars);
    console.log(`This React App is running on ${Port.REACT_APP_PORT}`);
    return (
      <div className="App">
        <h1 className="Header">React Wars</h1>
        <div className="cardsFlow">
          {this.state.starwarsChars.map((item, index) => {
            return <Card somestuff={item} key={index} />;
          })}
        </div>
      </div>
    );
  }
}

export default App;
