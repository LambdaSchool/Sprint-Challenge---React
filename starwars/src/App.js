import React, { Component } from 'react';
import './App.css';
import Card from './Card.js'

class App extends Component {
  constructor() {
    super();
    this.state = {
      starwarsChars: [],
      prev: null,
      next: null,
      page: 'https://swapi.co/api/people',
    };
    this.goToNextPage = this.goToNextPage.bind(this);
    this.goToPreviousPage = this.goToPreviousPage.bind(this);
  }
  componentDidMount() {
    // feel free to research what this code is doing.
    // At a high level we are calling an API to fetch some starwars data from the open web.
    // We then take that data and resolve it our state.
    fetch( this.state.page )
      .then(res => {
        return res.json();
      })
      .then(data => {
        this.setState({
          prev: data.previous,
          next: data.next,
        });
        return Promise.all(data.results.map(async (info, i) => {
          data.results[i].homeworld = await fetch( info.homeworld )
            .then(res => res.json())
            .then(home => home.name );
          return data.results[i];
        }));
      })
      .then(data => {
        this.setState({
          starwarsChars: data,
        });
      })
      .catch(err => {
        throw new Error(err);
      });
  }
  goToNextPage() {
    this.setState({ page: this.state.next, }, () => { this.componentDidMount() });
  }
  goToPreviousPage() {
    this.setState({ page: this.state.prev, }, () => { this.componentDidMount() });
  }
  render() {
    return (
      <div className="App">
        <nav className='nav-Header'>
          <button onClick={ this.goToPreviousPage } disabled={ !(this.state.prev) }>previous</button>
          <h1 className="Header">React Wars</h1>
          <button onClick={ this.goToNextPage } disabled={ !(this.state.next) }>next</button>
        </nav>
        <content>
          {this.state.starwarsChars.map(char => {
            return <Card key={ char.url.replace(/[^0-9]/g, '') } char={ char }/>
          })}
        </content>
      </div>
    );
  }
}

export default App;
