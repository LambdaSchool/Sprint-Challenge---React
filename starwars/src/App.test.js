import React from "react";
import App from './App';
import renderer from "react-test-renderer";

describe("<App />", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("<App />", () => {
    it("renders without crashing", () => {
      const tree = (
        <div className="App">
          <h1>React Wars</h1>
          <h1>React Wars - Star Wars</h1>
        </div>
      );
  
      expect(tree).toMatchSnapshot();
    });
  });