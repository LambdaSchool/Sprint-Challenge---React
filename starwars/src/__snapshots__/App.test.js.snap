// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
<div
  className="App"
>
  <h1>
    React Wars
  </h1>
  <h1>
    React Wars - Star Wars
  </h1>
  <div
    className="sc-ifAKCX dHcTOt"
  />
</div>
`;

exports[`<App /> renders without crashing 2`] = `
<div
  className="App"
>
  <h1>
    React Wars
  </h1>
  <h1>
    React Wars - Star Wars
  </h1>
</div>
`;
